packages/core/src/codewhisperer/commands/startTransformByQ.ts

```diff
@@ -232,6 +233,39 @@ export async function finalizeTransformByQ(status: string) {
     }
 }
 
+export async function parseBuildFile() {
+    try {
+        const absolutePaths = ['users/', 'system/', 'volumes/', 'c:', 'd:']
+        const alias = path.basename(os.homedir())
+        absolutePaths.push(alias)
+        const buildFilePath = path.join(transformByQState.getProjectPath(), 'pom.xml')
+        if (fs.existsSync(buildFilePath)) {
+            const buildFileContents = fs.readFileSync(buildFilePath).toString().toLowerCase()
+            const detectedPaths = []
+            for (const absolutePath of absolutePaths) {
+                if (buildFileContents.includes(absolutePath)) {
+                    detectedPaths.push(absolutePath)
+                }
+            }
+            if (detectedPaths.length > 0) {
+                const warningMessage = CodeWhispererConstants.absolutePathDetectedMessage
```

Comment:
Please use string interpolation here: https://stackoverflow.com/questions/45399951/how-to-perform-string-interpolation-in-typescript. We shouldn't need to be replacing magic strings in text

Reply:
Yes. Additionally, you can get rid of this new constant `CodeWhispererConstants .absolutePathDetectedMessage` and add the message directly to the `AbsolutePathDetectedError` class. There, you can pass your params and autofill the message with string interpolation

`new AbsolutePathDetectedError({numPaths: 3, paths: ['my/path'], buildFile: 'my/buildfile.file']})`

Reply:
If I use string interpolation, my String constant is just going to be floating around in `startTransformByQ.ts` rather than being stored in the constants file, so I'd prefer to keep it in the constants file unless you think this is highly problematic. And I already do add the message directly to the `AbsolutePathDetectedError` class if you see a few lines below.

Reply:
Why cant it live in constants anymore if you use string interpolation? It should be fine to keep it there. Or alternatively what max said works as well

Reply:
I need to write out the String in-line no? Something like `We detected {numPaths} file paths in file {fileName}` which won't allow me to keep the String as a constant in that file

Reply:
Not sure if i'm understanding 😅 but why not? There are several instances of doing this in the constants file:
e.g. https://github.com/aws/aws-toolkit-vscode/blob/master/packages/core/src/codewhisperer/models/constants.ts#L167, https://github.com/aws/aws-toolkit-vscode/blob/master/packages/core/src/codewhisperer/models/constants.ts#L169, https://github.com/aws/aws-toolkit-vscode/blob/master/packages/core/src/codewhisperer/models/constants.ts#L179

Reply:
```typescript
export const absolutePathDetectedMessage = (numPaths: number, buildFile: string, listOfPaths: string[]) =>
    'I detected ${numPaths} absolute file path(s) in your ${buildFile} file: **${listOfPaths}**. Absolute file paths might cause issues when I build your code. Any errors will show up in the build log.'
```

Reply:
Ah now I see. I didn't realize string interpolation could be done that way too, fixing now.